src/Showcase/src/Smart.FA.Catalog.Showcase.Web/Pages/Training/TrainingList/TrainingList.cshtml.cs

```diff
@@ -1,49 +1,43 @@
 #nullable disable
 using Microsoft.AspNetCore.Mvc;
-using Microsoft.EntityFrameworkCore;
 using Smart.FA.Catalog.Shared.Collections;
-using Smart.FA.Catalog.Shared.Domain.Enumerations.Training;
-using Smart.FA.Catalog.Showcase.Web.Mappers;
+using Smart.FA.Catalog.Showcase.Web.Services.Training;
 
 namespace Smart.FA.Catalog.Showcase.Web.Pages.Training.TrainingList;
 
 public class TrainingListModel : PageModelBase
 {
-    private readonly Infrastructure.Data.CatalogShowcaseContext _context;
+    private readonly ITrainingService _trainingService;
 
     public PagedList<TrainingListViewModel> Trainings { get; set; }
 
     [BindProperty(SupportsGet = true)] public int CurrentPage { get; set; } = 1;
 
     const int ItemsPerPage = 5;
 
-    public TrainingListModel(Infrastructure.Data.CatalogShowcaseContext context)
+   public TrainingListModel(ITrainingService trainingService)
     {
-        _context = context;
+        _trainingService = trainingService;
     }
-
-    public async Task<ActionResult> OnGetAsync()
+    public async Task<ActionResult> OnGetAsync([FromQuery] string? searchKeyword)
     {
         if (CurrentPage <= 0)
         {
-            return RedirectToNotFound("Page not found", "The page you have requested does not exist.");
+            return RedirectToNotFound();
         }
-        var offset = (CurrentPage - 1) * ItemsPerPage;
-
-        var trainingIdQuery = _context.TrainingList
-            .Where(training => training.Status == TrainingStatusType.Published.Id)
-            .Select(training => training.Id)
-            .Distinct();
-        var totalTrainerTrainings = await trainingIdQuery.CountAsync();
-
-        var paginatedIds = trainingIdQuery.OrderBy(trainingId => trainingId).Skip(offset).Take(ItemsPerPage);
-        var trainingList = await _context.TrainingList.Where(training => paginatedIds.Contains(training.Id)).ToListAsync();
 
-        var trainingListViewModel = trainingList.ToTrainingListViewModels();
+        Trainings = await _trainingService.SearchTrainingViewModelsAsync(searchKeyword, CurrentPage, ItemsPerPage);
```

Comment:
Any reason to include ViewModel in the function name?
```suggestion
        Trainings = await _trainingService.SearchTrainingsAsync(searchKeyword, CurrentPage, ItemsPerPage);
```